web_tls.go

```diff
@@ -68,6 +69,9 @@ func getCertificate(info *tls.ClientHelloInfo) (*tls.Certificate, error) {
 
 func getTLSState() TLSState {
 	s := TLSState{}
+
+	s.Enforce = config.TLSEnforce
```

Comment:
Move this down below the mode and cert setup and make it _also_ matter that the the config is valid.

Something like 
```go
canEnforce := false

switch config.TLSMode {
	case "disabled":
		s.Mode = "disabled"
	case "custom":
		s.Mode = "custom"
		cert := certStore.GetCertificate(webSecureCustomCertificateName)
		if cert != nil {
			var certPEM []byte
			// convert to pem format
			for _, c := range cert.Certificate {
				block := pem.Block{
					Type:  "CERTIFICATE",
					Bytes: c,
				}

				certPEM = append(certPEM, pem.EncodeToMemory(&block)...)
                canEnforce = true
			}
			s.Certificate = string(certPEM)

		}
	case "self-signed":
		s.Mode = "self-signed"
        canEnforce = true
	}

s.Enforce = canEnforce && config.TLSEnforce
```